src/msw/textctrl.cpp

```diff
@@ -1198,6 +1198,56 @@ void wxTextCtrl::SetRTFValue(const wxString& val)
     SetInsertionPoint(0);
 }
 
+wxTextSearchResult wxTextCtrl::SearchText(const wxTextSearch& search) const
+{
+    // set up the flags
+    WPARAM flags = 0;
+    if (search.m_direction == wxTextSearchDirection::Down)
+    {
+        flags |= FR_DOWN;
+    }
+    if (search.m_wholeWord)
+    {
+        flags |= FR_WHOLEWORD;
+    }
+    if (search.m_matchCase)
+    {
+        flags |= FR_MATCHCASE;
+    }
+
+    FINDTEXTEX findText;
+    findText.chrg.cpMin = (search.m_startingPosition != -1) ?
+        // user-provided start
+        search.m_startingPosition :
+        // if going down, then start from 0; otherwise, start from end
+        (search.m_direction == wxTextSearchDirection::Down) ? 0 : GetLastPosition();
+    if (search.m_direction == wxTextSearchDirection::Down)
+    {
+        // go to the end of the text
+        findText.chrg.cpMax = -1;
+    }
+    else
+    {
+        // if at the beginning of the window, then we can't search up
```

Comment:
Is this worth explicitly testing for? I.e. won't the native control just return "not found" anyhow?

Not a big deal, of course, but it seems strange to test for this but not that we're at the end when searching forward.

Reply:
Got it, removed that check.